app/src/main/java/com/example/hackernews/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.hackernews.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.hackernews.model.entities.User
+import com.example.hackernews.model.repository.UserRepository
+
+class LoginViewModel(private val userRepository: UserRepository) : ViewModel() {
+
+    private val _isRegisterSuccessfull = MutableLiveData(false)
+    private val _registerErrors = MutableLiveData<List<String>>()
+    // List is immutable, string is also, but I am not sure in combination with mutable live data it gives a immutable variable?
```

Comment:
`MutableLiveData` is a mutable class and it holds a single generic parameter that can be of any type. So when you write _registerErrors.value = some_list you are not mutating the underlying list but instead replace the reference with a new one. 

so for example _registerErrors.value.add() won't work as list is immutable